src/components/PDFView/index.js

```diff
@@ -1,46 +1,91 @@
-    style: PropTypes.any,
-};
-
-const defaultProps = {
-    sourceURL: '',
-    style: {},
-};
-
-class PDFView extends PureComponent {
+class PDFView extends Component {
     constructor(props) {
         super(props);
         this.state = {
             numPages: null,
             windowWidth: Dimensions.get('window').width,
+            shouldRequestPassword: false,
+            isPasswordInvalid: false,
         };
         this.onDocumentLoadSuccess = this.onDocumentLoadSuccess.bind(this);
+        this.initiatePasswordChallenge = this.initiatePasswordChallenge.bind(this);
+        this.attemptPdfLoad = this.attemptPdfLoad.bind(this);
+        this.avoidKeyboard = this.avoidKeyboard.bind(this);
     }
 
     /**
-     * Callback to be called to set the number of pages on PDF
+     * Upon successful document load, set the number of pages on PDF,
+     * hide/reset PDF password form, and notify parent component that
+     * user input is no longer required.
      *
      * @param {*} {numPages} No of pages in the rendered PDF
      * @memberof PDFView
      */
     onDocumentLoadSuccess({numPages}) {
-        this.setState({numPages});
+        this.setState({
+            numPages,
+            shouldRequestPassword: false,
+            isPasswordInvalid: false,
+        });
+    }
+
+    /**
+     * Initiate password challenge process. The react-pdf/Document
+     * component calls this handler to indicate that a PDF requires a
+     * password, or to indicate that a previously provided password was
+     * invalid.
+     *
+     * The PasswordResponses constants used below were copied from react-pdf
+     * because they're not exported in entry.webpack.
+     *
+     * @param {*} callback Callback used to send password to react-pdf
+     * @param {Number} reason Reason code for password request
+     */
+    initiatePasswordChallenge(callback, reason) {
+        this.onPasswordCallback = callback;
+
+        if (reason === CONST.PDF_PASSWORD_FORM.REACT_PDF_PASSWORD_RESPONSES.NEED_PASSWORD) {
+            this.setState({shouldRequestPassword: true});
+        } else if (reason === CONST.PDF_PASSWORD_FORM.REACT_PDF_PASSWORD_RESPONSES.INCORRECT_PASSWORD) {
+            this.setState({shouldRequestPassword: true, isPasswordInvalid: true});
+        }
+    }
+
+    /**
+     * Send password to react-pdf via its callback so that it can attempt to load
+     * the PDF.
+     *
+     * @param {String} password Password to send via callback to react-pdf
+     */
+    attemptPdfLoad(password) {
+        this.setState({isPasswordInvalid: false});
+        this.onPasswordCallback(password);
+    }
+
+    /**
+     * On Android browsers notify parent that the UI should be updated to
+     * accommodate keyboard.
+     *
+     * @param {Boolean} shouldAvoidKeyboard
+     */
+    avoidKeyboard(shouldAvoidKeyboard) {
+        if (getOperatingSystem() !== CONST.OS.ANDROID) {
+            return;
+        }
```

Comment:
Well, this condition will never run, and using this way of platform matching is prohibited in components.

Reply:
That's incorrect. It does run. In any case, this code is here to hide the send button on android-mweb. According to the mockups the send button should not be visible when the keyboard is open.

On iOS the send button is nicely covered by the keyboard. On Android, however, the send button gets squished up against the commit button when the keyboard is open.

I'll change this to hide the send button on all narrow screens. I don't think this will have any impact on iOS since the send button is covered anyway.

Reply:
We can try to stick with common behavior as much as possible. For example, here hiding the button regardless of the platform.